### ISSUE TITLE
Consider adding `steps` to `slsa-provenance`

### ISSUE THREAD

**Issue by priyawadhwa** [slsa-framework/slsa#177]
We have a `steps` field separate from `invocation` in `tekton-provenance`, and I think it clearly separates what started the build from what happened in the build. Especially if we do #175 , I think this would be nice to have as well. Opening this for discussion.

**Comment by TomHennen** [comment]
Can you document why it would be nice to have this as a top level(?) field independent from recipe/invocation?  I.e. why not just define a custom recipe/invocation type and specify the steps in the arguments field?

**Comment by priyawadhwa** [comment]
Yup, so this would go alongside the change described [here](https://github.com/slsa-framework/slsa/issues/175#issuecomment-937008279) where we would rename `recipe` to `invocation`, which corresponds to "what kicked off the build". 

We could then have a `recipe.steps` field (as we do in tekton) or just a `steps` field, which would correspond to "what happened during the build".

**Comment by joshuagl** [comment]
As Tom suggested, `recipe.type` is a [`TypeURI`](https://github.com/in-toto/attestation/blob/main/spec/field_types.md#TypeURI) and thus this _could_ be handled with a custom arguments object.

Did Tekton decide that was undesirable when designing the Tekton predicate?

**Comment by dlorenc** [comment]
> As Tom suggested, `recipe.type` is a [`TypeURI`](https://github.com/in-toto/attestation/blob/main/spec/field_types.md#TypeURI) and thus this _could_ be handled with a custom arguments object for the Tekton `recipe.type`.

I think this was changed after the Tekton format got created, but I can't remember the exact timing.
